core/src/main/scala/org/apache/spark/serializer/SerializerManager.scala

```diff
@@ -181,7 +181,7 @@ private[spark] class SerializerManager(
       blockId: BlockId,
       values: Iterator[_],
       classTag: ClassTag[_]): ChunkedByteBuffer = {
-    val bbos = new ChunkedByteBufferOutputStream(1024 * 1024 * 4, ByteBuffer.allocate)
+    val bbos = new ChunkedByteBufferOutputStream(chunkSize, ByteBuffer.allocate)
```

Comment:
Previously it is a fixed size. When you use a configurable value converted from long to int, doesn't it increase the chance to overflow?